Unicycler fixes

bebatut opened the pull request:
Hi,

I worked a bit on the wrapper of Unicycler

- Fix some options that were not correctly taken into account in the command
- Format a bit (mostly the indentation in the command section)
- Add default values to some params and remove their "optionality"
- Fix paired collection
- Move some params to sections

Berenice

mvdbeek commented:
> Add default values to some params and remove their "optionality"

I don't know unicycler well enough to have a strong opinion on this, but I think it would be good to mirror the typical command line usage. I.e skip unnecessary default arguments. That way you can go more easily from command line to gaIaxy and back.

There are other programs - like minimap2 and bwa, where setting a default is actually very confusing, because based on the preset you choose the default is altered, and there is no one "default" setting.

bebatut commented:
I tried to mirror the command line usage. I moved all the advanced options to the sections, but there I changed the `optional='true' help='default=sth'` to `value='sth'` and removing the optional parameter if a default value is given in Unicycler doc (otherwise I kept the optinality).
I hope it is more clear and that is ok (otherwise I can revert the changes)

bebatut commented on tools/unicycler/unicycler.xml:
```diff
@@ -1,188 +1,114 @@
+    </macros>
     <requirements>
-         <requirement type="package" version="0.4.1">unicycler</requirement>
+         <requirement type="package" version="@WRAPPER_VERSION@">unicycler</requirement>
```
Changed

bebatut commented:
@mvdbeek do you think I should remove all the default values and re-add the optional?

mvdbeek commented:
It's up to you, I'm happy to merge as is.

bebatut commented:
If you think it is better for long-term, I can change it. We should also add that in the best practices, maybe

mvdbeek commented:
I think it would be better long-term, but I guess it's not as bad as in minimap2/bwa, and we don't have an explicit rule on this anyway. We did discuss this at the GCC, but I don't remember the consensus :/. I don't want to block your work, as I guess it would take some time to have a result for the IUC standards.

nsoranzo commented:
@bebatut I've checked the default values you set and compared them with the Unicycler help, everything looks good to me.
